### ISSUE TITLE
validate: verify UID/GID for workflow environment images 

### ISSUE THREAD

**Issue by tiborsimko** [reanahub/reana-client#457]
**Current behaviour**

Some workflows are forcing certain Kubernetes user IDs, for example in `reana-demo-atlas-recast`:

```
$ cat workflow/steps.yml
...
  environment:
    environment_type: 'docker-encapsulated'
    image: reanahub/reana-demo-atlas-recast-eventselection
    imagetag: '1.0'
    resources:
      - kubernetes_uid: 500
```

This can become troublesome if the image is not using the same UID by default, or if GID is not 0.  If either of these is not matched, then the job may not have write rights, and it fails.

If this happens during late in the execution, this is basically waisting time.

It would be good to catch these situations faster, i.e. at the `reana-client validate --environment` time.

Currently, we verify this manually as follows:

```console
$ docker pull reanahub/reana-demo-atlas-recast-eventselection:1.0
$ docker run -i -t --rm reanahub/reana-demo-atlas-recast-eventselection:1.0 /usr/bin/id
uid=500(atlas) gid=500(atlas) groups=500(atlas),0(root)
```

This is good for two reasons: UID matches to what is declared in steps yaml; and even if it would not, then GID is 0, ensuring write permissions.

**Expected behaviour**

After we introduce `validate --environment` flags, see e.g. https://github.com/reanahub/reana-client/issues/382#issuecomment-781976172, we could enrich it to catch these situations.

Ideally, verify UID/GID without downloading image, but I think that is not easily possible... So this should be an extra option.

Perhaps using `singularity` or `podman` could help... The same considerations as in the base issue.

Basically, this issue would take the validation even further -- not only validating image existence, but also its UID and GID setup.

**Comment by tiborsimko** [comment]
Meta-remark: this issue is about the validation on the client side, continuing to building up on the easier checks.  However, we could also think of having a new endpoint and do the environment validation (i.e. docker run etc) on the server side. This could also help with restricted images, as it would verify whether the platform has well access or not.

Let's develop some Python code snippets doing described validation, and let's see later whether we plug some into `reana-client`, some into a new endpoint, and reana-client would just call these on demand.

**Comment by mvidalgarcia** [comment]
Closed by https://github.com/reanahub/reana-client/pull/461
